src/json.hpp

```diff
@@ -1535,17 +1489,18 @@ class input_buffer_adapter : public input_adapter_protocol
     {
         if (JSON_LIKELY(cursor < limit))
         {
-            return *(cursor++) & 0xFF;
+            return reinterpret_cast<int>(std::char_traits<char>::to_int_type(*(cursor++)));
```

Comment:
This seems to fail on MSVC, see https://ci.appveyor.com/project/nlohmann/json/build/2302/job/n1ti8cx5qqafk43i.

Reply:
a `static_cast` should suffice here